lib/segment/src/index/struct_payload_index.rs

```diff
@@ -309,7 +310,26 @@ impl StructPayloadIndex {
             }
             Condition::Field(field_condition) => self
                 .estimate_field_condition(field_condition, nested_path)
-                .unwrap_or_else(|| CardinalityEstimation::unknown(self.available_point_count())),
+                .unwrap_or_else(|| {
```

Comment:
TBH I am unsure if it is the best spot for generating the issues :thinking: 

The cardinality estimation is used to pick the driving filter from the query, that is the one with the highest selectivity.

I am worried this generates a lot of issue objects under high read load with lots of filters.
Most of those will be discarded because they exist already in the in-memory issues store. 

An alternative would be to place it in the function `condition_converter` to catch the actual field driving the query.

Can you maybe elaborate on why you picked this approach? :)

Reply:
While I agree that this might not be the best place for generating the issues. I would argue that it is more because it might have side effects for when the cardinality estimation is not directly used for searching or filtering, rather than if the issue is trying to be generated for every unindexed condition. 

My rationale is: if the search is slow already due to unindexed field, it won't affect much to try to do a read operation on the issues dashboard (when checking if the issue already exists).

After seeing if I can move this generation to the `condition_converter` function, I realized there is no easy way to include the collection name without drilling it down across many parent functions. So the question is: would it be worth it to increase the complexity for this?

Reply:
Another place we could add it is in:
https://github.com/qdrant/qdrant/blob/7ac7398bb129946c7c770f882272a0917d647237/lib/segment/src/index/struct_payload_index.rs#L456-L460

So that only the primary conditions trigger the issue

cc @generall